compiler/packages/react-mcp-server/src/tools/runtimePerf.ts

```diff
@@ -96,38 +122,67 @@ export async function measurePerformance(
     await page.waitForFunction(
       'window.__RESULT__ !== undefined && (window.__RESULT__.renderTime !== null || window.__RESULT__.error !== null)',
     );
+
     // ui chaos monkey
-    await page.waitForFunction(`window.__RESULT__ !== undefined && (function() {
-      for (const el of [...document.querySelectorAll('a'), ...document.querySelectorAll('button')]) {
-        console.log(el);
-        el.click();
+    const selectors = await page.evaluate(() => {
+      window.__INTERACTABLE_SELECTORS__ = [];
+      const elements = Array.from(document.querySelectorAll('a')).concat(
+        Array.from(document.querySelectorAll('button')),
+      );
+      for (const el of elements) {
+        window.__INTERACTABLE_SELECTORS__.push(el.tagName.toLowerCase());
       }
-      return true;
-    })() `);
-    const evaluationResult: PerformanceResults = await page.evaluate(() => {
+      return window.__INTERACTABLE_SELECTORS__;
+    });
+
+    for (const selector of selectors) {
+      await page.click(selector);
+      await delay(500);
+    }
```

Comment:
depending on the number of elements this could be pretty slow. I don't think you need to wait 500ms after every click before clicking again right?

```suggestion
    await Promise.all(selectors.map((selector) => page.click(selector)));
    await delay(500);
```